src/lightning_app/core/flow.py

```diff
@@ -634,3 +643,81 @@ def my_remote_method(self, name):
             lightning my_command_name --args name=my_own_name
         """
         raise NotImplementedError
+
+    def state_dict(self):
+        """Returns the current flow state but not its children."""
```

Comment:
I wouldn't use the same names as PyTorch. 

1. They don't do the same. Here the children are explicitly excluded
2. In PyTorch normally you don't override the state_dict and load_state_dict methods directly, instead you override [get_extra_state](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.set_extra_state) and set_extra_state. Here, it is the opposite: You are supposed to override them but not call them directly. 

For these reasons, I suggest to call them a different name. For example: `state_without_children` and `load_state`.